react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { currentIndex: 0 };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(event) {
+    if (event.target.closest('div').classList.contains('arrow-left')) {
+      this.setState({ currentIndex: ((this.state.currentIndex + this.props.items.length) - 1) % this.props.items.length });
+      this.componentWillUnmount();
+    } else if (event.target.closest('div').classList.contains('arrow-right')) {
+      this.setState({ currentIndex: ((this.state.currentIndex + this.props.items.length) + 1) % this.props.items.length });
+      this.componentWillUnmount();
+    } else if (event.target.closest('ul').classList.contains('index')) {
+      this.setState({
+        currentIndex: (event.target.tagName === 'li')
+          ? parseInt(event.target.id, 10)
+          : parseInt(event.target.closest('li').id, 10)
+      });
+      this.componentWillUnmount();
+    }
+    this.componentDidMount();
+  }
```

Comment:
Instead of having "complex" if statements checking for DOM elements, you can make 3 separate methods and just use those on the appropriate elements. It will simplify your code.